src/ckb-daemon/structures.h

```diff
@@ -247,7 +248,7 @@ typedef struct {
     // Color dithering in use
     char dither;
     // Flag to check, if large macros should be sent delayed
-    char delay;
+    uint delay;
```

Comment:
I'm on a Mac, using Xcode's gcc to compile, UINT_MAX was 4-billion something. 

I did try recording a delayed macro, on my version, not yours, and hitting the key a few times. There was a long delay but the macro played back and the keys afterwards were queued and played. The macro was not as long as yours, and I think I hit it maybe 3 to 4 times, not 10 to 20.

I suspect the key buffer is getting overflowed causing the disconnect. Perhaps there should be a limit to the delay time? I think this would only make the problem harder to reproduce rather than fixing it though.

The question is, what to do when this happens? Queue up more macro keys (make the queue larger), drop some of them (at the start or end of the queue), limit the queue size, or ignore it and pretend it doesn't happen, pretend it doesn't crash.

1. Making the queue larger might actually make the keyboard seem broken to people. If a really long macro is playing back and they hit keys, they won't see anything happening, their keys will be delayed until earlier ones finish playing.

2. Dropping keys seems tricky. First, which keystrokes to drop? Ones at the start or end of the queue? I could see making the case for either and for both of them being the wrong choice.

3. Limiting the queue size would be the same as dropping the most recent keys struck.

4. Ignoring the problem is kind of what happens now. Not sure that's the right thing to do.

I'm leaning towards limiting the queue size as being the right answer here (4). Figuring out when the queue is full and ignoring new input. Maybe there should there be command that cancels macro playback? One you could bind to another key or in a sequence.

This is getting tricky.